src/block/block_ext.c

```diff
@@ -1332,6 +1332,24 @@ __wti_block_extlist_write(
     return (ret);
 }
 
+/*
+ * __wti_block_extlist_can_truncate --
+ *     Check if there is an available extent at the end of the file that can be truncated.
+ */
+bool
+__wti_block_extlist_can_truncate(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_EXTLIST *el)
+{
+    WT_EXT **astack[WT_SKIP_MAXDEPTH], *ext;
+
+    if ((ext = __block_off_srch_last(el->off, astack)) == NULL)
+        return (false);
+    WT_ASSERT(session, ext->off + ext->size <= block->size);
+    if (ext->off + ext->size < block->size)
+        return (false);
```

Comment:
Suggesting a few comments here:

```suggestion
    /* Retrieve the last available extent. */
    if ((ext = __block_off_srch_last(el->off, astack)) == NULL)
        return (false);

    /* The extent should not go beyond the boundaries of the file. */
    WT_ASSERT(session, ext->off + ext->size <= block->size);

    /* The extent retrieved does not cover the end of the file. */
    if (ext->off + ext->size < block->size)
        return (false);
```

Reply:
Added https://github.com/wiredtiger/wiredtiger/pull/11176/commits/9c619d686ebf950af87e898e5b2fece60e5bb6a8